### PR TITLE
Prevent noop index settings updates

### PR DESCRIPTION
**Pull request by vknabel**
Due to https://github.com/meilisearch/meilisearch-go/issues/452 disabling an index' typo tolerance is ineffectful. Therefore the index settings migration will always run. So we skip this check.

In the other cases we did a sorted comparisions of the slices, but received a different order from meilisearch. Therefore an unsorted comparison prevents excessive index updates.

### CODE REVIEW

File: auditing/meilisearch.go [hunk 1]
```diff
@@ -503,3 +507,20 @@ func indexName(prefix string, i Interval) string {
 	indexName := prefix + "-" + time.Now().Format(timeFormat)
 	return indexName
 }
+
+func slicesUnorderedEqual[T comparable](lhs, rhs []T) bool {
```
Comment (Gerrit91) [inline auditing/meilisearch.go:511]: Can you provide a unit test for this?

Comment (vknabel) [reply auditing/meilisearch.go:511]: Sure. As it's currently unexported, I either:
* can export it withing auditing (though it does not really belong there)
* need to move it to a different module where it makes sense (`pkg/multisort`?)
* might just write a test in the same `auditing` package to be able to access it

Neither felt right. What do you think? I slightly prefer moving to multisort.

Comment (Gerrit91) [reply auditing/meilisearch.go:511]: Not sure if we should export it. For me it's ok here in this package only and unit tested.

File: auditing/meilisearch.go [hunk 2]
```diff
@@ -400,20 +400,24 @@ func (a *meiliAuditing) migrateIndexSettings(index *meilisearch.Index) error {
...
 		changesRequired = true
 		diff.SortableAttributes = desired.SortableAttributes
 	}
-	if !slices.Equal(current.SearchableAttributes, desired.SearchableAttributes) {
+	if !slicesUnorderedEqual(current.SearchableAttributes, desired.SearchableAttributes) {
 		changesRequired = true
 		diff.SearchableAttributes = desired.SearchableAttributes
 	}
-	if !slices.Equal(current.FilterableAttributes, desired.FilterableAttributes) {
+	if !slicesUnorderedEqual(current.FilterableAttributes, desired.FilterableAttributes) {
```
Comment (majst01) [inline auditing/meilisearch.go:420]: Maybe a dump question: Why not simply sort both slices before and use the slices.Equal ?

There is sort.Slice available

Comment (vknabel) [reply auditing/meilisearch.go]: The updated version does it 👍

**Review by majst01 — APPROVED** [review]

**Review by Gerrit91 — APPROVED** [review]

**Comment by Gerrit91** [comment]
Unfortunately, there is a regression in go-testcontainers in combination with go 1.20.6: https://github.com/testcontainers/testcontainers-go/issues/1359

For the tests to pass, you can specifically downgrade to 1.20.5 in the CI action.

**Comment by majst01** [comment]
> Unfortunately, there is a regression in go-testcontainers in combination with go 1.20.6: [testcontainers/testcontainers-go#1359](https://github.com/testcontainers/testcontainers-go/issues/1359)
> 
> For the tests to pass, you can specifically downgrade to 1.20.5 in the CI action.

Update to `github.com/docker/docker v24.0.5+incompatible ` as transitive dependency fixes this

**Pull request merged**